### PR TITLE
Wait till netexec is ready in kubeproxy e2e

### PR DESCRIPTION
**Pull request by bprashanth**
Use a simple readiness probe to wait till the udp server is listening on 8081

### CODE REVIEW

**Comment by bprashanth** [comment]
e2e will fail because i haven't pushed the image out

**Comment by thockin** [comment]
LGTM overall, though it might be worth polishing and pushing this image for wider use.

Version doesn't need 3 fields.  1.x is easier to manage.
Probably should just be on port 80
Probably could have a better name :)
Could simplify the URL args (e.g. /shell?cmd=date)
Could support dial?protocol=tcp in addition to http/udp

**Comment by thockin** [comment]
@k8s-bot test this please

**Comment by bprashanth** [comment]
I'd ask k8s bot to test this, but it looks like we have jenkins instability right now. Current failure looks like a kubectl proxy flake.

**Comment by thockin** [comment]
@k8s-bot test this please

**Comment by bprashanth** [comment]
@k8s-bot test this please

File: test/images/netexec/netexec.go [hunk 1]
```diff
@@ -283,6 +310,9 @@ func startUDPServer(udpPort int) {
 	defer serverConn.Close()
 	buf := make([]byte, 1024)
 
+	log.Printf("Started udp server")
+	// Start responding to readiness probes.
+	serverReady.set()
```
Comment (ArtfulCoder) [inline test/images/netexec/netexec.go]: can you add a defer that resets serverReady.
that way, if udp server dies, the health check will fail.

**Comment by ArtfulCoder** [comment]
@bprashanth do you plan to check this in. if not lets close it.
I believe the e2e tests are already using an image based on the change ? In that case, we should get this PR merged. I just had one comment and a rebase is required.

**Comment by ArtfulCoder** [comment]
@bprashanth ping

**Comment by bprashanth** [comment]
@ArtfulCoder PTAL, should help with https://github.com/kubernetes/kubernetes/issues/20552

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #19500

File: test/images/netexec/netexec.go [hunk 2]
```diff
@@ -318,6 +351,10 @@ func startUDPServer(udpPort int) {
 	defer serverConn.Close()
 	buf := make([]byte, 1024)
 
+	log.Printf("Started udp server")
```
Comment (ArtfulCoder) [inline test/images/netexec/netexec.go]: can you add a log statement after the for to indicate the udp server stopped.

**Comment by bprashanth** [comment]
@ArtfulCoder repushed, LETS GET THIS IN! :)

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #22655

**Pull request merged**